plugins/package-managers/conan/src/main/kotlin/PackageInfo.kt

```diff
@@ -56,3 +74,120 @@ internal data class PackageInfoV1(
     val requires: List<String> = emptyList(),
     val buildRequires: List<String> = emptyList()
 ) : PackageInfo
+
+@Serializable
+internal data class PackageInfoV2(
+    override val author: String?,
+    override val revision: String?,
+    override val url: String?,
+    val ref: String? = null,
+    val license: List<String> = emptyList(),
+    val homepage: String? = null,
+    val label: String,
+    val requires: List<String> = emptyList(),
+    val buildRequires: List<String> = emptyList(),
+    val recipeFolder: String? = null,
+    val description: String? = null,
+    // The next two properties can be null for the first package, i.e. the conanfile itself. To simplify the processing
+    // by the package manager, they are exposed as non-nullable properties.
+    val name: String,
+    val version: String,
+    val binaryRemote: String? = null,
+    val packageType: PackageType
+) : PackageInfo
+
+@Serializable
+private data class PackageInfoV2Raw(
```

Comment:
Same question here, is the split between `PackageInfoV2` and `PackageInfoV2Raw` really needed?

Reply:
As you can see, the properties are wrapped into a `graph.nodes` map. Having `PackageInfoV2Raw` allow to unwrap them later to the `PackageInfoV2` structure, the latter being easier to use in the handler.